src/services/DidService.ts

```diff
@@ -18,6 +19,23 @@ const { IS_IN_BLOCK } = BlockchainUtils
 class DidService {
   public static readonly URL = `${window._env_.REACT_APP_SERVICE_HOST}/contacts/did`
 
+  public static async fetchDID(
+    identity: Identity
+  ): Promise<IDidDocumentSigned | null> {
+    const did = await Did.queryByAddress(identity.address)
+    if (did) {
+      const didDocument = Did.createDefaultDidDocument(
+        did.identifier,
+        did.publicBoxKey,
+        did.publicSigningKey,
+        did.documentStore || undefined
```

Comment:
Unfortunately, this is not meant by this param. It wants the kilt service endpoint, which is not stored on-chain.
You will have to fetch the `did.documentStore` url, verify its signature and use its contents as the did document.